WCFDataService/Client/System/Data/Services/Client/MaterializeFromAtom.cs

```diff
@@ -69,6 +69,11 @@ internal class MaterializeAtom : IDisposable, IEnumerable, IEnumerator
         private bool moved;
 
         /// <summary>
+        /// The ODataResponseMessage to dispose when we are done.
+        /// </summary>
+        private IODataResponseMessage responseMessage;
```

Comment:
The idea is to keep the response message (or more specifically the http connection) alive during enumeration so the materializing process can read the data directly from the wire. If I dispose of the response message (and thus closing the http connection) after ExecuteQuery() have been called (like in the $count code path), the enumeration will fail as it cannot read from the closed http connection.

Could you elobarate on what the failing case specifically is?